src/sentry/models/group.py

```diff
@@ -201,10 +202,14 @@ def get_oldest_or_latest_event_for_environments(
         )
         dataset = Dataset.Transactions
     else:
+        if group.issue_category == GroupCategory.ERROR:
+            dataset = Dataset.Events
+        else:
+            dataset = Dataset.IssuePlatformGeneric
```

Comment:
so we want the fallback `Dataset` to be generic? feel like it should stay `Events` for now.

also I'd clean up this logic with an `ifelse`

Reply:
Hmm, I think that if a transaction ends up here then we're in a bad spot either way, since neither dataset is correct.

Not totally sure i understand about the `elif`